worlds/factorio/Options.py

```diff
@@ -104,9 +113,57 @@ class FactorioWorldGen(OptionDict):
                                       "stone": {"frequency": 1, "size": 3, "richness": 6},
                                       "trees": {"frequency": 1, "size": 1, "richness": 1},
                                       "uranium-ore": {"frequency": 1, "size": 3, "richness": 6}},
-               "starting_area": 1, "peaceful_mode": False,
+               "seed": None,
+               "starting_area": 1,
+               "peaceful_mode": False,
                "cliff_settings": {"name": "cliff", "cliff_elevation_0": 10, "cliff_elevation_interval": 40,
-                                  "richness": 1}}
+                                  "richness": 1},
+               "pollution": {"enabled": True, "diffusion_ratio": 0.02, "ageing": 1,
+                             "enemy_attack_pollution_consumption_modifier": 1,
+                             "min_pollution_to_damage_trees": 60,
+                             "pollution_restored_per_tree_damage": 10}}
+    schema = Schema({
+        "basic": {
+            Optional("terrain_segmentation"): FloatRange(0.166,6),
+            Optional("water"): FloatRange(0.166,6),
+            Optional("autoplace_controls"): {
+                str: {
+                    "frequency": FloatRange(0,6),
+                    "size": FloatRange(0,6),
+                    "richness": FloatRange(0.166,6)}},
+            Optional("seed"): Or(None,And(int, lambda n: n>=0)),
+            Optional("starting_area"): FloatRange(0.166,6),
+            Optional("peaceful_mode"): LuaBool,
+            Optional("cliff_settings"): {
+                "name": str, "cliff_elevation_0": FloatRange(0,99),
+                "cliff_elevation_interval": FloatRange(0.066,241),  # 40/frequency
+                "richness": FloatRange(0,6)},
+        },
+        "advanced": {
+            Optional("pollution"): {
+                Optional("enabled"): LuaBool,
+                Optional("diffusion_ratio"): FloatRange(0,0.25),
+                Optional("ageing"): FloatRange(0.1,4),
```

Comment:
For this, and other percentage based options, isn't `0` a reasonable minimum? Factorio allows for setting 0%, right?

Reply:
I checked all the values in Factorio GUI when I wrote the code.
Most values seem to be from 1/6 to 6x `FloatRange(0.166,6)`, some have a special 0 (i.e. disable; setting anything 0 and 0.17 will just give you the 17%). Pollution is special in the meaning of its values, so they have different limits in the GUI. Unless i screwed up, the ranges should match the limits in the GUI.